api_admin_panel/app/components/query.py

```diff
@@ -0,0 +1,193 @@
+            rx.chakra.accordion_panel(
+                rx.match(
+                    QueryState.current_req,
+                    (
+                        "GET",
+                        rx.select(
+                            QueryState.get_params_body,
+                            default_value="None",
+                            width="100%",
+                        ),
+                    ),
+                    (
+                        "POST",
+                        rx.vstack(
+                            rx.hstack(
+                                item_add_event(event_trigger),
+                                width="100%",
+                                justify_content="end",
+                            ),
+                            rx.select(
+                                QueryState.post_params_body,
+                                default_value="JSON",
+                                width="100%",
+                            ),
+                            rx.vstack(
+                                rx.foreach(state, func), width="100%", spacing="1"
+                            ),
+                            width="100%",
+                        ),
+                    ),
+                ),
+            ),
+        ),
+        allow_toggle=True,
+        width="100%",
+    )
+
+
+def form_request_item():
+    return rx.chakra.accordion(
+        rx.chakra.accordion_item(
+            rx.chakra.accordion_button(item_title("Requests")),
+            rx.chakra.accordion_panel(
+                rx.hstack(
+                    rx.select(
+                        QueryState.req_methods,
+                        width="120px",
+                        default_value="GET",
+                        on_change=QueryState.get_request,
+                    ),
+                    rx.input(
+                        value=QueryState.req_url,
+                        width="100%",
+                        on_change=QueryState.set_req_url,
+                        placeholder="https://example_site.com/api/v2/endpoint.json",
+                    ),
+                    width="100%",
+                )
+            ),
+        ),
+        allow_toggle=True,
+        width="100%",
+    )
+
+
+def render_query_form():
+    return rx.vstack(
+        form_request_item(),
+        # form_item("URL Parameters"),
+        form_item(
+            "Headers", QueryState.headers, form_item_entry, QueryState.add_header
+        ),
+        form_body_param_item(QueryState.body, form_item_entry, QueryState.add_body),
+        form_item(
+            "Cookies", QueryState.cookies, form_item_entry, QueryState.add_cookies
+        ),
+        width="100%",
+        spacing="0",
+        padding="0.5em 0.75em",
+    )
+
+
+def render_query_header():
+    return rx.hstack(
+        rx.button("Run", size="2", on_click=QueryAPI.run_get_request),
+        width="100%",
+        border_bottom="1px solid rgba(41, 41, 41, 0.51)",
```

Comment:
This border looks good in dark mode in light mode looks a little off may need to use rx.color shade instead